igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -17,55 +17,41 @@
 
 package com.netflix.spinnaker.igor.build
 
-import com.fasterxml.jackson.databind.ObjectMapper
 import com.netflix.spinnaker.igor.build.model.GenericBuild
 import com.netflix.spinnaker.igor.exceptions.BuildJobError
 import com.netflix.spinnaker.igor.exceptions.QueuedJobDeterminationError
 import com.netflix.spinnaker.igor.jenkins.client.model.JobConfig
 import com.netflix.spinnaker.igor.jenkins.service.JenkinsService
-import com.netflix.spinnaker.igor.model.BuildServiceProvider
 import com.netflix.spinnaker.igor.service.ArtifactDecorator
+import com.netflix.spinnaker.igor.service.BuildProperties
 import com.netflix.spinnaker.igor.service.BuildService
 import com.netflix.spinnaker.igor.service.BuildServices
-import com.netflix.spinnaker.kork.core.RetrySupport
+import com.netflix.spinnaker.igor.travis.service.TravisService
 import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException
 import com.netflix.spinnaker.kork.web.exceptions.NotFoundException
 import groovy.transform.InheritConstructors
 import groovy.util.logging.Slf4j
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.web.bind.annotation.*
 import org.springframework.web.servlet.HandlerMapping
-import org.yaml.snakeyaml.Yaml
-import org.yaml.snakeyaml.constructor.SafeConstructor
 import retrofit.RetrofitError
 
 import javax.servlet.http.HttpServletRequest
-import java.util.concurrent.ExecutorService
 
 import static net.logstash.logback.argument.StructuredArguments.kv
 import static org.springframework.http.HttpStatus.NOT_FOUND
 
 @Slf4j
 @RestController
 class BuildController {
-
-    @Autowired
-    ExecutorService executor
-
-    @Autowired
-    BuildServices buildMasters
-
     @Autowired
-    ObjectMapper objectMapper
-
-    @Autowired
-    RetrySupport retrySupport
+    private BuildServices buildMasters
 
     @Autowired(required = false)
-    BuildArtifactFilter buildArtifactFilter
+    private BuildArtifactFilter buildArtifactFilter
 
     @Autowired(required = false)
-    ArtifactDecorator artifactDecorator
+    private ArtifactDecorator artifactDecorator
```

Comment:
We tend to like classes to have the pattern of an autowired constructor instead of autowired fields, even though a lot of the older code has the autowired field pattern.

```
private BuildArtifactFilter buildArtifactFilter
private ArtifactDecorator artifactDecorator

@Autowired
public BuildController(BuildArtifactFilter buildArtifactFilter, ArtifactDecorator artifactDecorator) {
this.buildArtifactFilter = buildArtifactFilter
...
```
To make this work with `@Autowired(required = false)` you just wrap the constructor param in an optional, and then do any checking/error handling/default values in the constructor assignment.

If you're already doing major refactoring and turning classes into java this might be another little goodie to throw in. Not necessary though!

Reply:
Good point, I sort of stopped here as this was getting to be a big PR, but I'm all for constructor injection so I'll add it in!